apis/pkg/meta/v1/provider_types.go

```diff
@@ -0,0 +1,135 @@
+	errWrongConvertFromProvider = "must convert from *meta.Provider"
+)
+
+// ProviderSpec specifies the configuration of a Provider.
+type ProviderSpec struct {
+	// Configuration for the packaged Provider's controller.
+	Controller ControllerSpec `json:"controller"`
+
+	MetaSpec `json:",inline"`
+}
+
+// ControllerSpec specifies the configuration for the packaged Provider
+// controller.
+type ControllerSpec struct {
+	// Image is the packaged Provider controller image.
+	Image string `json:"image"`
+
+	// PermissionRequests for RBAC rules required for this provider's controller
+	// to function. The RBAC manager is responsible for assessing the requested
+	// permissions.
+	// +optional
+	PermissionRequests []rbacv1.PolicyRule `json:"permissionRequests,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+
+// A Provider is the description of a Crossplane Provider package.
+type Provider struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ProviderSpec `json:"spec"`
+}
+
+// ConvertTo converts this Provider to the Hub version.
+func (p *Provider) ConvertTo(hub conversion.Hub) error {
+	out, ok := hub.(*meta.Provider)
+	if !ok {
+		return errors.New(errWrongConvertToProvider)
+	}
+
+	p.ObjectMeta.DeepCopyInto(&out.ObjectMeta)
+
+	out.Spec = meta.ProviderSpec{
+		Controller: meta.ControllerSpec{
+			Image:              p.Spec.Controller.Image,
+			PermissionRequests: p.Spec.Controller.PermissionRequests,
+		},
+	}
+
+	if p.Spec.Crossplane != nil {
+		out.Spec.Crossplane = &meta.CrossplaneConstraints{Version: p.Spec.Crossplane.Version}
+	}
+
+	if len(p.Spec.DependsOn) == 0 {
+		return nil
+	}
+
+	out.Spec.DependsOn = make([]meta.Dependency, len(p.Spec.DependsOn))
+	for i := range p.Spec.DependsOn {
+		out.Spec.DependsOn[i] = meta.Dependency{
+			Provider:      p.Spec.DependsOn[i].Provider,
+			Configuration: p.Spec.DependsOn[i].Configuration,
+			Version:       p.Spec.DependsOn[i].Version,
+		}
+	}
+
+	return nil
+}
+
+// ConvertFrom converts this Provider from the Hub version.
+func (p *Provider) ConvertFrom(hub conversion.Hub) error {
+	in, ok := hub.(*meta.Provider)
+	if !ok {
+		return errors.New(errWrongConvertFromProvider)
+	}
+
+	in.ObjectMeta.DeepCopyInto(&p.ObjectMeta)
+
+	p.Spec = ProviderSpec{
+		Controller: ControllerSpec{
+			Image:              in.Spec.Controller.Image,
+			PermissionRequests: in.Spec.Controller.PermissionRequests,
+		},
+	}
+
+	if in.Spec.Crossplane != nil {
+		p.Spec.Crossplane = &CrossplaneConstraints{Version: in.Spec.Crossplane.Version}
+	}
+
+	if len(in.Spec.DependsOn) == 0 {
+		return nil
+	}
+
+	p.Spec.DependsOn = make([]Dependency, len(in.Spec.DependsOn))
+	for i := range in.Spec.DependsOn {
+		p.Spec.DependsOn[i] = Dependency{
+			Provider:      in.Spec.DependsOn[i].Provider,
+			Configuration: in.Spec.DependsOn[i].Configuration,
+			Version:       in.Spec.DependsOn[i].Version,
+		}
+	}
```

Comment:
Good point, but I'd like to address this in a future PR. That way we can take a little more time to establish type conversion patterns, since we'll presumably be doing this a fair bit once we introduce conversion webhooks.